aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -211,6 +227,34 @@ namespace {
     }
     return descriptors;
   }
+#else
+  auto rnn_descriptor_sequence(const Tensor& tensor, const int batch_size, IntArrayRef batch_sizes, const int seq_len, const int vector_size) { // packed case
+    RNNDataDescriptor r;
+    std::vector<int> seqLengthArray(batch_size, 1);
+    // TODO(eqy): There's probably a smarter way to do this than O(SN)
+    for (auto it = batch_sizes.begin(); it != batch_sizes.end(); it++) {
+      // everyone starts at sequence length 1 so we skip an iteration
```

Comment:
What does this mean? That wasn't there in the old code

Reply:
Added a comment to correspond to the packed batch explanation---cuDNN wants the per-sequence sequence lengths of a packed batch as if they were unpacked. It saves having to create the vector of `TensorDescriptor`s that we had to do previously, but the trade is that now we have to provide this additional metadata.